packages/toolkit/src/extensionNode.ts

```diff
@@ -7,15 +7,48 @@ import type { ExtensionContext } from 'vscode'
 import { activate as activateCore, deactivate as deactivateCore } from 'aws-core-vscode/node'
 import { awsToolkitApi } from './api'
 import { Commands } from 'aws-core-vscode/shared'
+import * as semver from 'semver'
+import * as vscode from 'vscode'
+import { telemetry } from 'aws-core-vscode/telemetry'
 
 export async function activate(context: ExtensionContext) {
     await activateCore(context)
 
     // after toolkit is activated, ask Amazon Q to register toolkit api callbacks
     await Commands.tryExecute('aws.amazonq.refreshConnectionCallback', awsToolkitApi)
+    void setupVscodeVersionNotification()
     return awsToolkitApi
 }
 
 export async function deactivate() {
     await deactivateCore()
 }
+
+// TODO: remove once version bump to 1.83.0 is complete.
+export function setupVscodeVersionNotification() {
```

Comment:
Do we need duplicate code?
AFAIK the null extension bug only impacts when running the extension in debug mode and not when its packaged into a vsix.

Reply:
Ah, not aware of that. I can remove the duplicate code in that case.